src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -259,3 +265,41 @@
                (map (fn [cause] [CauseViewer cause]) causes)])
             (when (seq stack-trace)
               [StackTraceViewer {:lines stack-trace}])]))))})
+
+
+(defn- validate-crumb [crumb]
+  (contains? crumb :text))
+
+(react/defc Breadcrumbs
+  {:push
+   (fn [{:keys [state]} new-crumb]
+     (assert (validate-crumb new-crumb) "Crumb must have :text")
+     (swap! state update-in [:crumbs] conj new-crumb))
+   :set-crumbs
+   (fn [{:keys [state]} crumbs]
+     (assert (every? validate-crumb crumbs))
+     (swap! state assoc :crumbs (vec crumbs)))
+   :get-initial-state
+   (fn [{:keys [props]}]
+     (assert (every? validate-crumb (:initial-crumbs props))
+       "Each initial crumb must have :text")
+     {:crumbs (vec (:initial-crumbs props))})
+   :render
+   (fn [{:keys [state]}]
+     (let [sep (icons/font-icon {:style {:fontSize "50%" :margin "0 0.5em"}} :angle-right)
+           crumbs (:crumbs @state)]
+       (case (count crumbs)
+         0 [:div {}]
+         1 [:div {} (:text (first crumbs))]
+         [:div {}
+          (interpose sep
+            (map-indexed
+              (fn [index {:keys [text on-click href]}]
+                (if (or on-click href)
+                  (style/create-link2 {:href href :text text
```

Comment:
validate-crumb checks that breadcrumbs must contain :text, but it looks like you're also counting on it to contain :href.